python/tank/descriptor/io_descriptor/git.py

```diff
@@ -147,6 +147,14 @@ def _clone_then_execute_git_commands(self, target_path, commands):
             log.debug("Restoring cwd (to '%s')" % cwd)
             os.chdir(cwd)
 
+        # Cleanup the folder of any git related files.
+        shutil.rmtree(os.path.join(target_path, ".git"), ignore_errors=True)
```

Comment:
I think the use case we were trying to cater for was:

- use (classic) setup project wiz to create a new project from a git repo
- once your project config is set up, you want it to still be an active git repo so that you can pull down changes from a remote branch etc. Note [https://github.com/shotgunsoftware/tk-core/blob/master/python/tank/descriptor/io_descriptor/git.py#L214](this comment) where we make sure that we copy the `.git` as the project setup copies the config from the bundle cache out to the config location.

I think this was changed in 0.18 - previously, the project setup was totally standalone and duplicated a bunch of the git code. In 0.18, everything was unified to use the ∂esccriptor API.

I agree that it is unnecessary to pull down the `.git` repo in most cases but i am afraid that we need this for the time being due to the case above. Maybe we can refactor the project setup to be smarter? We can create a separate ticket to look at that (but i doubt it will be a high prio).

Does this make sense? Does it align with the current logic?

If so, it would be great if you could add some comments to the code to explain to future readers. Apologies for not doing this in the first place.